test/parallel_api/iterator/indirectly_device_accessible.pass.cpp

```diff
@@ -0,0 +1,464 @@
+    friend bool
+    operator==(const base_strided_iterator& a, const base_strided_iterator& b)
+    {
+        return a.base == b.base;
+    }
+
+    friend bool
+    operator!=(const base_strided_iterator& a, const base_strided_iterator& b)
+    {
+        return !(a == b);
+    }
+
+    friend bool
+    operator<(const base_strided_iterator& a, const base_strided_iterator& b)
+    {
+        return a.base < b.base;
+    }
+
+    friend bool
+    operator<=(const base_strided_iterator& a, const base_strided_iterator& b)
+    {
+        return a.base <= b.base;
+    }
+
+    friend bool
+    operator>(const base_strided_iterator& a, const base_strided_iterator& b)
+    {
+        return a.base > b.base;
+    }
+
+    friend bool
+    operator>=(const base_strided_iterator& a, const base_strided_iterator& b)
+    {
+        return a.base >= b.base;
+    }
+
+  private:
+    BaseIter base;
+    int stride;
+};
+
+template <typename BaseIter>
+struct first_strided_iterator : public base_strided_iterator<BaseIter>
+{
+    first_strided_iterator(BaseIter base, int stride) : base_strided_iterator<BaseIter>(base, stride) {}
+};
+
+template <typename BaseIter>
+auto
+is_onedpl_indirectly_device_accessible_iterator(first_strided_iterator<BaseIter>)
+{
+    return oneapi::dpl::is_indirectly_device_accessible<BaseIter>{};
+}
+
+template <typename BaseIter>
+struct second_strided_iterator : public base_strided_iterator<BaseIter>
+{
+    second_strided_iterator(BaseIter base, int stride) : base_strided_iterator<BaseIter>(base, stride) {}
+};
+
+template <typename BaseIter>
+auto
+is_onedpl_indirectly_device_accessible_iterator(second_strided_iterator<BaseIter>)
+    -> decltype(oneapi::dpl::is_indirectly_device_accessible<BaseIter>{});
+
+template <typename BaseIter>
+struct third_strided_iterator : public base_strided_iterator<BaseIter>
+{
+    third_strided_iterator(BaseIter base, int stride) : base_strided_iterator<BaseIter>(base, stride) {}
+    friend auto
+    is_onedpl_indirectly_device_accessible_iterator(third_strided_iterator<BaseIter>)
+    {
+        return oneapi::dpl::is_indirectly_device_accessible<BaseIter>{};
+    }
+};
+
+template <typename BaseIter>
+struct fourth_strided_iterator : public base_strided_iterator<BaseIter>
+{
+    fourth_strided_iterator(BaseIter base, int stride) : base_strided_iterator<BaseIter>(base, stride) {}
+    friend auto
+    is_onedpl_indirectly_device_accessible_iterator(fourth_strided_iterator<BaseIter>)
+        -> oneapi::dpl::is_indirectly_device_accessible<BaseIter>;
+};
+
+} // namespace custom_user
+
+template <bool base_indirectly_device_accessible, typename BaseIter>
+void
+test_with_base_iterator()
+{
+    //test assumption about base iterator device accessible content iterator
+    static_assert(oneapi::dpl::is_indirectly_device_accessible_v<BaseIter> == base_indirectly_device_accessible,
+                  "is_onedpl_indirectly_device_accessible_iterator is not working correctly for base iterator");
```

Comment:
It is normal that we check `oneapi::dpl::is_indirectly_device_accessible_v` but in the assert message we have another name - `is_onedpl_indirectly_device_accessible_iterator`?